elide-core/src/main/java/com/yahoo/elide/security/FilterExpressionCheck.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2016, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.security;
+
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.security.checks.InlineCheck;
+import java.util.Optional;
+
+/**
+ * Check for FilterExpression
+ *
+ * @param <T> Type of class
+ */
+public abstract class FilterExpressionCheck<T> extends InlineCheck<T> {
+
+    public abstract FilterExpression getFilterExpression(RequestScope requestScope);
+
+    @Override
+    public boolean ok(T object, RequestScope requestScope, Optional<ChangeSpec> changeSpec) {
```

Comment:
I wonder if this should evaluate the FilterExpression in memory returned from getFilterExpression.  Also, maybe make this final so the behavior can't be changed to something that would break assumptions.